client/src/components/categorical/categorical.js

```diff
@@ -176,10 +176,18 @@ class Categories extends React.Component {
           }
         />
 
+        {/* Categories with only one field */}
+        <div style={{marginLeft: 0}}>
+          {allCategoryNames.map(catName => 
```

Comment:
can you break this out into a helper function.  This kind of inline code is hard to follow.